packages/core/src/messageformat.ts

```diff
@@ -70,6 +70,26 @@ export interface MessageFormatOptions<
         };
   };
 
+  /**
+   * If defined, used by {@link compileModule} to identify and map keys to
+   * the locale identifiers used by plural rules.
+   *
+   * Default: `undefined`
+   *
+   * @example
+   * ```js
+   * // Support all recognised Unicode locale identifiers
+   * function pluralCodeFromKey(key) {
+   *   try {
+   *     return new Intl.Locale(key).language
+   *   } catch {
+   *     return null
+   *   }
+   * }
+   * ```
+   */
+  pluralCodeFromKey?: ((key: string) => string | null | undefined) | null;
```

Comment:
I propose `getLocaleCodeFromKey` as an improved name.

We are indexing into a `plurals[localeCode]` internally--indeed! However my original ticket/failure had nothing to do with plurals, thus I was confused why I had to provide a `pluralCodeFromKey`. "To fix my ICU date formatting issue, why would I need to meddle with plurals?"

I studied the source, and now understand. `plurals` is _indexed_ by locale, and ...locale is kind of a superset of pluralCode. To hinge all locale formatting off of the plural code is a fine choice, but users of the MessageFormat API think in terms of locales, not pluralCodes (an implementation detail), so it would be great to keep that tucked inside.

`getLocaleCodeFromKey` as the interface communicates to me: "Hey you, function author, map over each of your object keys and derive the correct locale for them. By the way, it must be compliant with pluralCodes, even if your ICU message has no plurals in it".

Are you open to this suggestion?

Reply:
The challenge with calling the function a "locale code getter" is that at least with the default plurals, you may want the _input_ to be a locale code like `es-MX` while the _output_ might be `es`.

Reply:
can you help clarify why that is problematic?

`getLocaleCodeFromKey`

input: `Key`
output: `LocaleCode`

key is an arbitrary description. Per your own design, the user controls what type of content the key is. `es` & `es-MX`--both valid locale codes, contingent on what a "code" is defined as. So long as the return goes through `plurals::normalize(...)`, shouldn't it be safe? Perhaps it's not being normalized. Perhaps we should always normalize the return value from this function

Reply:
hey @eemeli, did you see my feedback?

let me know if you want me to send a patch. otherwise, we can just take the patches as you have authored them. i'm eager to get the next release :)

thanks for your time & energy!

Reply:
hey @eemeli, sorry to pester! i know you're busy. i'd rather cut a release with what you have than drag out anything further. please kindly let me know if there's anything i can do to help the solution proceed :)